lib/hyacinth/client.rb

```diff
@@ -0,0 +1,254 @@
+      query = <<~GQL
+        mutation ($input: UpdateResourceRequestInput!) {
+          updateResourceRequest(input: $input) {
+            resourceRequest {
+              id
+            }
+          }
+        }
+      GQL
+
+      variables = {
+        input: {
+          id: resource_request_id,
+          status: status,
+          processingErrors: processing_errors
+        }
+      }
+
+      graphql(query, variables).tap do |gql_response|
+        gql_response['errors'].tap do |errors|
+          raise Hyacinth::Client::Exceptions::UnexpectedResponse, "Errors encountered during update_resource_request attempt: #{errors.inspect}" if errors.present?
+        end
+      end
+
+      true
+    end
+
+    def update_featured_thumbnail_region(digital_object_uid, featured_thumbnail_region)
+      query = <<~GQL
+        mutation ($input: UpdateFeaturedThumbnailRegionInput!) {
+          updateFeaturedThumbnailRegion(input: $input) {
+            digitalObject {
+              id
+            }
+            userErrors {
+              message
+            }
+          }
+        }
+      GQL
+
+      variables = {
+        input: {
+          id: digital_object_uid,
+          featuredThumbnailRegion: featured_thumbnail_region
+        }
+      }
+
+      graphql(query, variables).tap do |gql_response|
+        gql_response['errors'].tap do |errors|
+          raise Hyacinth::Client::Exceptions::UnexpectedResponse, "Errors encountered during update_featured_thumbnail_region attempt: #{errors.inspect}" if errors.present?
+        end
+        gql_response.dig('data', 'updateFeaturedThumbnailRegion', 'userErrors').tap do |user_errors|
+          raise Hyacinth::Client::Exceptions::UnexpectedResponse, "Errors encountered during create_resource attempt: #{user_errors.inspect}" if user_errors.present?
+        end
+      end
+
+      true
+    end
+
+    def graphql(query, variables = {})
+      params = { query: query }
+      params[:variables] = variables.is_a?(Hash) ? variables.to_json : variables unless variables.blank?
+
+      response = @conn.post('/graphql') do |req|
+        req.params = params
+      end
+
+      JSON.parse(response.body)
+    end
+
+    # TODO: Maybe delete this method since Digest::MD5.file(file_path) seems to be equivalent and just as fast (from limited benchmarking).
+    # Calculates the md5 digest of the given file, reading the file in buffered chunks to keep
+    # memory usage low during calculation.
+    def self.buffered_read_md5_digest(file_path, buffer_size = 1.megabyte)
+      md5digest = ::Digest::MD5.new
+      File.open(file_path, 'rb') do |f|
+        while (chunk = f.read(buffer_size))
+          md5digest.update(chunk)
+        end
+      end
+      md5digest
+    end
+
+    module Exceptions
+      class Error < StandardError; end
+      class UnexpectedResponse < Error; end
+    end
+
+    # Note: The method below is not currently in use, but may be useful later.
```

Comment:
@JackBlackLight Just a heads up that unlike the unused method above, I actually would like to leave this one below.  There may be some things that we learn after deployment and use that could necessitate something like this.